lychee-lib/src/remap.rs

```diff
@@ -1,20 +1,39 @@
+//! Remapping rules which allow to map URLs matching a pattern to a different
+//! URL.
+//!
+//! # Notes
+//! Use in moderation as there are no sanity or performance guarantees.
+//!
+//! - There is no constraints on remapping rules upon instantiation or during
+//!   remapping. In particular, rules are checked sequentially so later rules
+//!   might contradict with earlier ones if they both match a URL.
+//! - A large rule set has a performance impact because the client needs to
+//!   match every link against all rules.
+
+// Notes on terminology:
+// The major difference between URI(Uniform Resource Identifier) and
+// URL(Uniform Resource Locator) is that the former is an indentifier for
+// resources and the later is a locator.
+// We are not interested in differentiate resources by names and the purpose of
+// remapping is to provide an alternative **location** in certain
+// circumanstances. Thus the documentation should be about remapping URLs
+// (locations), not remapping URIs (identities).
+
 use std::ops::Index;
 
-use crate::{ErrorKind, Result};
 use regex::Regex;
 use reqwest::Url;
 
-use crate::Uri;
+use crate::ErrorKind;
 
-/// Remaps allow mapping from a URI pattern to a specified URI
+/// Rules that remaps URL matching patterns.
 ///
-/// Some use-cases are
-/// - Testing URIs prior to production deployment
-/// - Testing URIs behind a proxy
+/// Some use-cases are:
+/// - Testing URLs prior to production deployment.
+/// - Testing URLs behind a proxy.
 ///
-/// Be careful when using this feature because checking every link against a
-/// large set of regular expressions has a performance impact. Also there are no
-/// constraints on the URI mapping, so the rules might contradict each other.
+/// # Notes
+/// See module level documentation of usage notes.
```

Comment:
```suggestion
/// See module-level documentation of usage notes.
```

Reply:
https://github.com/search?q=repo%3Arust-lang%2Frust+%22module+level%22&type=code

they are two words. there are plenty of "module level documentation" everywhere.